### DISCUSSION TITLE
Simulation of patient flows and service use in a hospital?

### DISCUSSION THREAD

**Discussion by JustGitting** [gama-platform/gama#51] [Q&A]
Hi,

I’ve been asked to help with a simulation modelling project to model the flow of patients through a hospital with multiple departments (ED, inpatient, etc) with various resources (number of doctors per ward, etc). I’ll be collaborating with experts who use AnyLogic, but I’m keen to look at FOSS alternatives.

I searched for examples of GAMA for this problem domain, but could not find anything obvious.

Is GAMA capable of modelling people/services/flows/resources within an organisation, such as a hospital? Have you got any examples of it’s use?

Thanks for your help.

**Comment by lesquoyb** [comment]
Hello, I'm quite sure gama can fit those requirements. I know @ptaillandier worked on similar models, maybe he can give some pointers/example models

**Comment by pszps** [comment]
This sounds more like a queuing process model, which would require a discrete event simulation implementation. If you are a Python programmer, you could use SimPy for building such models. For Java programmers there is DesmoJ (but it's quite dated).

**Reply by JustGitting** [reply]
Thanks @pszps, I'd prefer python. 

Is it possible to convert AnyLogic models to an open-standard for SimPy for example?

**Comment by AlexisDrogoul** [comment]
Hi, the main question is what your model is supposed to explain / explore / solve as you will get different answers for different tasks. if it is about optimising the flows and the resources use, in a pure OR approach, then queuing models are better adapted than any ABM representations including GAMA. However, if the interaction between people, their behaviour (following rules or not), or the (re)organisation of space plays a role at some point in what you need to explore, then an ABM is more adapted. Same if you need to have some kind of interaction with users of the model or if your model is supposed to be integrated, i.e. to also serve other purposes (for instance, @ptaillandier worked on this topic but regarding the impact on virus propagation, which means at least two different models were interacting). 
And in that cas, we'd be happy to explore with you the possibilities offered by GAMA !

**Reply by JustGitting** [reply]
Thanks @AlexisDrogoul, 

The aim will be to model patient flows (wait times, etc) and the impact that increasing/decreasing staff numbers (taking into there are doctors, nurses, specialist, etc) Also it would need to consider external resource constrains, such as waiting for a free bed in inpatient wards, or the time taken to have an X-ray for example.

**Reply by AlexisDrogoul** [reply]
Hi. OK. Sounds like pure OR then. Not sure you need to go ABM. Good luck with your project !

**Reply by JustGitting** [reply]
What foss platform would your recommend for OR?

**Reply by AlexisDrogoul** [reply]
No idea. But it seems that the domain is a bit lagging behind others in terms of open-source platform (e.g. https://or.stackexchange.com/questions/7526/why-is-open-source-operations-research-software-so-far-behind-open-source-statis)

**Reply by pszps** [reply]
You cannot translate AnyLogic models into SimPy models automatically. Regarding free open source OR simulation, you could try this one: https://jaamsim.com/. Not sure what the differences are between the free and the pro version as I just spotted it.

**Reply by JustGitting** [reply]
Hi @AlexisDrogoul,

Just to clarify, if we want to model the ED physical layout, trying different layouts, then ABM would be needed (e.g. GAMA)? Or only if there are interactions between people?

**Reply by AlexisDrogoul** [reply]
I think that one of the advantages of ABM is that you can take into account individual variabilities in behavior, choices, following the rules or not, and also learning. It is then possible to model individuals who -- like humans -- are diverse and evolve over time thanks to their interactions and exchanges. It is in my opinion crucial to better understanding social phenomena, but your needs really depend on the goal of your project. 
Regarding the physical environment I guess that many OR tools can take geographical/layout constraints into account, but I'm not expert enough to tell you which :)
